staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -108,6 +109,42 @@ func (c *cache) put(s string, a *Authenticator) *Authenticator {
 	return a
 }
 
+// sometimes rate limits how often a function f() is called. Specifically, Do()
+// will run the provided function f() up to threshold times every interval
+// duration.
+//
+// Note that this type is not thread-safe, so ya gotta do your own locking.
+type sometimes struct {
+	threshold int
+	interval  time.Duration
+
+	count  int       // times we have called f() in this window
```

Comment:
These need to be synchronized?

Reply:
I think the reason why I didn't put a lock in `sometimes` is because I assumed it would be OK if it transitively used the synchronization mechanisms provided as a part of the `Authenticator`. WDYT?

Reply:
I went ahead and added synchronization in dd8cc37d5251c. I'm still interested in your thoughts in order to learn the best pattern here, but I don't feel strongly about my above design decision.